src/lib/containers/table/datasets/DatasetItemsEditor.tsx

```diff
@@ -216,10 +212,20 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
             versionNumber: item.targetVersionNumber!,
           })),
         ]
-        return {
+
+        const updatedDataSet = {
           ...datasetToUpdate,
           items: items,
         }
+        displayToast(SAVE_THE_DATASET_TO_CONTINUE, 'info', {
+          title: toastMessageTitle,
+          primaryButtonConfig: {
+            text: 'Save changes to Draft',
+            onClick: () => mutation.mutate(updatedDataSet),
+          },
+        })
+
+        return updatedDataSet
```

Comment:
Added displayToast inside setDatasetToUpdate was that when setting as mutation.mutate(datasetToUpdate) the state isn't updated when the button is clicked. Also, the toast title depends on the data within setDatasetToUpdate.

Reply:
Could you do something like
```tsx
function fireUpdate() {
	mutation.mutate(datasetToUpdate)
}

function addItemsToDataset(itemsToAdd: Reference[]) {
	if (datasetToUpdate) {
		let toastMessageTitle = ''
		setDatasetToUpdate(datasetToUpdate => {
			// existing logic to update state
			// update toastMessageTitle as needed
		})
		displayToast(SAVE_THE_DATASET_TO_CONTINUE, 'info', {
			title: toastMessageTitle,
			primaryButtonConfig: {
				text: 'Save changes to Draft',
				onClick: fireUpdate,
			},
		})
	} // unchanged from here
}


```